### PR TITLE
Add AccountLinesRPC unit tests.

### PR DESCRIPTION
**Pull request by scottschurr**
Reviewers: @miguelportilla, @nbougalis

### CODE REVIEW

**Comment by vinniefalco** [comment]
HELLO +++Coverage!!!

**Comment by miguelportilla** [comment]
I left some comments. Overall excellent comprehensive test that improves coverage. Nice job, Scott!

**Comment by scottschurr** [comment]
Addressed review comments.  Thanks for the pointers @miguelportilla.

File: src/ripple/rpc/handlers/AccountLines.cpp [hunk 1]
```diff
@@ -97,13 +97,8 @@ Json::Value doAccountLines (RPC::Context& context)
     std::string strIdent (params[jss::account].asString ());
     AccountID accountID;
 
-    if (auto jv = RPC::accountFromString (accountID, strIdent))
-    {
-        for (auto it = jv.begin (); it != jv.end (); ++it)
-            result[it.memberName ()] = it.key ();
-
+    if ((result = RPC::accountFromString (accountID, strIdent)))
```
Comment (nbougalis) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Extraneous parens here.

File: src/ripple/rpc/handlers/AccountLines.cpp [hunk 2]
```diff
@@ -116,10 +111,7 @@ Json::Value doAccountLines (RPC::Context& context)
     AccountID raPeerAccount;
     if (hasPeer)
     {
-        result[jss::peer] = context.app.accountIDCache().toBase58 (accountID);
-        result = RPC::accountFromString (raPeerAccount, strPeer);
-
-        if (result)
+        if ((result = RPC::accountFromString (raPeerAccount, strPeer)))
```
Comment (nbougalis) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Same here - no need for the extra pair of parens.

**Comment by nbougalis** [comment]
Just a comment about seemingly extra parens and assignment inside an `if`. I don't need any changes - it's fine as is. :+1:

**Comment by miguelportilla** [comment]
:+1:

File: src/ripple/rpc/handlers/AccountLines.cpp (hunk 1 above)
Comment (scottschurr) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Yeah.  The extra parentheses were to remove this warning from clang:
```
warning: using the result of an
      assignment as a condition without parentheses [-Wparentheses]
    if (result = RPC::accountFromString (accountID, strIdent))
```
Probably the smart thing to do here is add a comment about the motivation for the extra parentheses.

Comment (vinniefalco) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Should we consider disabling the warning? Assignment in conditional when introducing a scoped variable seems to be getting popular.

Comment (scottschurr) [inline src/ripple/rpc/handlers/AccountLines.cpp]: @vinniefalco This is a different case.  It's an assignment to a pre-existing variable in an `if`.  I think the warning serves a useful purpose.  And the extra parentheses silence the warning.

**Comment by scottschurr** [comment]
Added two comments and squashed.

**Comment by nbougalis** [comment]
Merged as acaf91a2f7c21e410f9159718287d0dc37cc12b3

**Pull request closed without merging**